cpp/models/abm/parameters.h

```diff
@@ -25,16 +25,48 @@
 #include "abm/virus_variant.h"
 #include "abm/vaccine.h"
 #include "abm/test_type.h"
+#include "memilio/config.h"
+#include "memilio/io/auto_serialize.h"
+#include "memilio/io/io.h"
+#include "memilio/math/time_series_functor.h"
 #include "memilio/utils/custom_index_array.h"
 #include "memilio/utils/uncertain_value.h"
 #include "memilio/utils/parameter_set.h"
 #include "memilio/epidemiology/age_group.h"
 #include "memilio/epidemiology/damping.h"
 #include "memilio/epidemiology/contact_matrix.h"
+
+#include <algorithm>
 #include <limits>
+#include <type_traits>
 
 namespace mio
 {
+
+template <class IOContext, class T,
+          std::enable_if_t<std::is_same_v<UniformDistribution<double>::ParamType, T>, void*> = nullptr>
+void serialize_internal(IOContext& io, const T& p)
```

Comment:
I assume you mean this function in pickle_serializer.h: 
```
template <class T, std::enable_if_t<PickleType<T>::value, void*> = nullptr>
friend void serialize_internal(PickleContext& io, const T& t)
```

My theory was that ParamType has a constructor `ParamType(double a, double b = 1.0)`, so there is a safe implicit conversion between int and ParamType. I tried to reproduce this in [godbolt](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZgQDbYB2AhgLbYgDkAjF%2BTXRMiAZVQtGIHgBYBQogFUAztgAKAD24AGfgCsp5eiyahUAUgBMAIUtXyKxqiIEh1ZpgDC6egFc2TEAA2C3J3ABkCJmwAOT8AI2xSEABmAFZyAAd0JWIXJi9ffyCQrJznIQio2LYEpLT7bEdyphEiFlIiAr8A4IamvNb2okqY%2BMSU9KU2jq6i3qmhkeraiYBKe3QfUlROLktkyNRfHABqM2SPH2d6YgBPc9wzLQBBfcPj7DOL1CnCYQenq8XlNSD4nCdVARUABrRgiRIECQEABeiTOAHYrBiACLnGwvQFEbBsDJGInnDxHFhKJQnAAqAOBRFB4NU7XYdNuGU%2BZkxZxeJ0FEPZbE53IgqwxNnRuIFQrZpA5XOwEDp5HpX2xJy0kt50tlrxleMBhOJpJY5O%2BRhp9PVVJt5y1ADd0ARMIzniCwUQIVDYZ8QCdfiAQDQJCoAPpEZVS3m45L455Eklk7AUj1e1l%2BxgUyJEB4nQPBkDMnzYKMxvVx40vZPmy0eDOlrMwnMXTCbOJt3CFoNETAh0vl6Pc2NGhOmlMWnlW6m0hnJR5Mlk%2ByGttMXNV9gcgZgsLvlgg0KMU4sEJQR9rEJAcZzmTcPENOiRlgGL3vFocV0dVgnjxN1qmXyUta84ej4OQmCc54RkobASPQEZ5tgwCKvQmrbiGyGoRIEYYEwCz/BccToN46oQGeF7YWhFILrgT4vjyFiBJYgRBii2DoDQqq6hcjonNIqxvgBZpARS9pgYugKZqu2Ybh4W7FnuB5Ice%2BYXMW%2BG6D4TBOHkEZOrmF5wQhSHCChNEPou6qURGOk5MAUTulZjxSW5PZFv2g6gsOlaYia/6TvWM6NlJy7er6665sIgTSCe76eTuX4jjy/l/vGiZBam4mgbamG7qwKlHvFHhrv6tGPiAz6%2BNg6oum6ABUBb8UwPj0PQGTMh69WYEGCJIqiZlEqQrD0BAZVwv1NyoqQrF2kIUz0qxQl/vqJq1qJ045XOJwAJIAPJeOZ6jqUuzw9X1pCItNh7mSNEgQAdR1Eidc0nPhi0Kkq3IUh2PgHgCzErYaa0Ei8eYnPBkQSr%2BzxCpF/rwldA1oveibw0oSBkD6Sg1nDQoQwQePw39B4nM56MEzFcU%2BvQxNCj4eaxVGJxtfTgpfaKyq/Z23YnBkeo8NWE5yoKKjIzdQ2JKNECoOqBBCSL%2BNi1NHFS/dY1y0GiuUyrEtq8hGuy%2BqmA64CGOq4Nhsy1r9Bm6Ll3XQbd02%2BqbX28rjso%2Brrv8x7cZcOs9DcKk/ABLw/DoNwjbWLYQabNsTHJHw5BENogfrEg2AsDgSQSuQ0IgKkqQAHQWNIAAcWjSDw6Loqk0ipIEACczGGNw0j8GwUhaFo5Dhzo5BR1w/BKCAfdp1wOjrHAsAoBgJIMIklDUAvGRL0kwA8DIAgMMNY8QHE6fkHEkTtLc3Ap6frCkLc%2B1xLo2BOJf/ALxwwj7Uw9AX1P/A4HEPhgAeAQmPCO5AcDwRMJIX%2B4CCCkCfs4J02BQGD2wOoJ%2BVxdgp2QsHGBNw4iKlvl4HAx9mQEG7mApBpASIqGxMSYwwAbgmHTusGgRhgBKAAGoEGwAAd32tycOKdBDCDEBITgO8RGKBUBoY%2B%2BgeCGAYWgWwthDAEDiGPWArAOC7gIMAZg5AqHcGSMkdEGch6dTyKAgAtL8R05hY7WFricaxAB1BCLjXFoOZCwTxiRSBkBcdyTAxg7zWP8WQJQkcqFXRwJo/ODgEF5DcEwTw3hugGHCJEUYNRxgKNKLkIQswAj5OyIUpgSwxhJAUYkvSQhBgzHSUUGpjQkn1OmMMbJyw8n2A6cUgwCwOiVNydU9YSgE47CkEHEOYdj7DxOOoCugRrGxROMAVAqATjbxLtIE4EB8DEECfsHgqx%2BCT2npnbOudqDrELtIaQJd648GbqkCuFhy4N2SFodEshcGd3IN3Hgvd%2B5zO4KPceqcWHTK4BYLuQRm5lx%2BaY2KtcLAmNSOiEFMDh5nJYeQWeyAQCbCIBkK4K9ZboEXowUg0R2C7EWcs1Z6zNnbOSPwbAhASCxIMFIsRkhJHyGUGoTQMD5HkF4YqDIL9oWhyxYPYe%2B0rikp9FxBZSyVm7OZVsngOy9leCpWiY5pzIW/1WJcnO4x85/LhV8x5Fgm5oteRYdEgRYqyAHpHMF9gIXnPMbgtlcrPUjxNRcwxiQciuGkEAA) but something is missing, I never get the ambiguous error.

Reply:
Here it is from the last "merge main" CI:
https://github.com/SciCompMod/memilio/actions/runs/10302043439/job/28514876106#step:4:275

Reply:
I tried the overload without enable_if locally and it worked. So I tried it in a branch and ran the CI and it worked as well, see here: 

commit https://github.com/SciCompMod/memilio/commit/88fbe2d2da6bfd5a4b47a311f9fa731be7b61dcc

CI run https://github.com/SciCompMod/memilio/actions/runs/10390162486

Is it possible this was resolved by your changes to PickleType or something? Could you check again?

(Sorry, I had to test this myself. I would be annoyed if this overload doesn't work, since this is the reason the design is the way that it is.)

Reply:
Ah, I did not realize that it is not treated as an error. You still get the warning in your branch: https://github.com/SciCompMod/memilio/actions/runs/10390162486/job/28769977939#step:4:279

Reply:
I didn't see that. I think we don't turn on warnings and warnings-as-errors in the bindings.

It's definitely not ideal to have this warning for portability. I'm just confused, since I didn't get it locally. I'll check again.

Reply:
I just missed the warning when I tried locally, since I install with `pip` and that only prints errors, no warnings. But it's there. I want to investigate this a little more, but maybe this doesn't need to hold up this merge request if the rest is done.